### DISCUSSION TITLE
Questions about preset and kAligner2

### DISCUSSION THREAD

**Discussion by jeongyeonkim95** [milaboratory/mixcr#1015] [Q&A] [answered]
Hello,
Thanks for developing wonderful tool.

I had a few questions and would really appreciate your insight.

1) I have BCR sequencing conducted by Mi-seq, which are not one of the presets.
In this case, would rnaseq-full-length be the most appropriate option?

2) I am trying to find somatic hypermutation in B cell, and the documentation recommends using 'kAligner2' for highly mutable sequences.
the kAligner2 seems to be deprecated in the newer version.
What would be an alternate approach to capture highly mutable sequence?

Thanks in advance

**Comment by mizraelson** [comment]
Hi,
Can you please describe your library structure/protocol in more detail? 
Also kAligner2 is still available and is the best choice for BCR data. I can help you in creating a specific preset (every preset includes an aligner option) if you can share the details of the library structure:
- is it a commercially available kit which has a description?

- is it an amplicon based protocol or rna seq (if it's an amplicon based based approach is it a 5'RACE or a primer multiplex or something else)

- do you have any barcodes (e.g. UMI, Cell etc.)? If the answer is yes - what's it structure?

**Comment by jeongyeonkim95** [comment]
Hi,
Thanks for the quick response.

I have tried command with no luck
mixcr align --species hs --preset kAligner2 test.fasta test2.vdjca

which returns 
'''
                    No preset with name "kAligner2".
                    Here are supported presets with similar names:
                    - generic-bcr-amplicon
- oncomine-human-igh-fr2-j
- generic-bcr-amplicon-umi
- oncomine-human-igh-leader-j
                    
                    To list all built-in presets run `mixcr listPresets`.
'''

It is commercially available kit : Illumina Mi-seq, generating 2x300bp paired end reads
Mi-seq is amplifies genomic DNA  with sequencing by synthesis (bridge) - same mechanism as HiSeq
I guess you could say that as typical Illumina DNA sequencing prep.
They have adaptors, which can be removed by adapter trimming ( guessing that this may be a necessary prestep)

Thanks again for providing insight into solving my issues

**Comment by mizraelson** [comment]
Hi, I'm sorry, I meant the kit or the protocol you used for cDNA library generation (I'm guessing you amplify BCRs using 5'RACE or multiplex), not the sequencing kit.

**Comment by jeongyeonkim95** [comment]
Oh yes, it was conducted with 5'RACE

**Comment by mizraelson** [comment]
And the reverse primer is located in the C gene? you don't have any UMI barcodes in the structure?

**Comment by jeongyeonkim95** [comment]
universal 5′ RACE primer IIA (Clontech), 5′AAGCAG TGGTATCAACGCAGAG 3′ 
and no UMI barcode

**Comment by mizraelson** [comment, marked as answer]
Got it, then I would recommend using the following command:

```
mixcr analyze generic-bcr-amplicon \
    --species hsa \
    --rna
    --rigid-left-alignment-boundary \
    --rigid-right-alignment-boundary C \
      input_R1.fastq.gz \
      input_R2.fastq.gz \
      result
```

Notice that you might have to change the name of the species.

This command will execute all of the steps needed including alignment, assemble and export.

Also, if you want to run `mixcr align` separately, you can use the preset `generic-bcr-amplicon`, which includes kAligner2.

Let me know if it helps.

**Comment by jeongyeonkim95** [comment]
Thanks so much!
I will give that command a try.
